### ISSUE TITLE
MapboxNavigation Upgrade Fails

### ISSUE THREAD

**Issue by zreaqqasem** [mapbox/mapbox-navigation-ios#3645]
### Mapbox Navigation SDK version

2.0.0

### Steps to reproduce

We are currently using MapboxNavigation version 0.40.0 and currently we are trying to upgrade MapboxNavigation to the latest version 2.0.0 using cocoapods.

### Expected behavior

the expected result was to install MapboxNavigation version successfully after change the version in podfille from 
 ` pod 'MapboxNavigation', '~> 0.40.0'`   to   `pod 'MapboxNavigation', '~> 2.0.0'`

### Actual behavior

the result I still get is -> 

`[!] CocoaPods could not find compatible versions for pod "MapboxMobileEvents":
  In snapshot (Podfile.lock):
    MapboxMobileEvents (= 0.10.2, ~> 0.10.2)

  In Podfile:
    MapboxNavigation (~> 2.0.0) was resolved to 2.0.1, which depends on
      MapboxMobileEvents (~> 1.0)

    react-native-mapbox-gl (from `../node_modules/@react-native-mapbox-gl/maps`) was resolved to 8.5.0, which depends on
      Mapbox-iOS-SDK (~> 5.9.0) was resolved to 5.9.0, which depends on
        MapboxMobileEvents (= 0.10.2)
        
        `

### Is this a one-time issue or a repeatable issue?

repeatable

**Comment by MaximAlien** [comment]
Hello @zreaqqasem, based on output you've provided both legacy `Mapbox-iOS-SDK` (linked implicitly via `react-native-mapbox-gl` library) and new `MapboxMaps` (linked implicitly via `MapboxNavigation`) libraries are used. 

Since they're using completely different versions of `MapboxMobileEvents`:
- `v1.0.6` of `MapboxMobileEvents` is used in the latest version of `MapboxNavigation` from `main` branch
- `v0.10.4` of `MapboxMobileEvents` is used in the latest version of `Mapbox-iOS-SDK` from `master` branch (`v0.10.2` in your case)

it causes dependencies clash.

Do you use `react-native-mapbox-gl` anywhere in your project at this point? If not I recommend removing it, it should fix your problem.

**Comment by zreaqqasem** [comment]
> Hello @zreaqqasem, based on output you've provided both legacy `Mapbox-iOS-SDK` (linked implicitly via `react-native-mapbox-gl` library) and new `MapboxMaps` (linked implicitly via `MapboxNavigation`) libraries are used.
> 
> Since they're using completely different versions of `MapboxMobileEvents`:
> 
> * `v1.0.6` of `MapboxMobileEvents` is used in the latest version of `MapboxNavigation` from `main` branch
> * `v0.10.4` of `MapboxMobileEvents` is used in the latest version of `Mapbox-iOS-SDK` from `master` branch (`v0.10.2` in your case)
> 
> it causes dependencies clash.
> 
> Do you use `react-native-mapbox-gl` anywhere in your project at this point? If not I recommend removing it, it should fix your problem.

hey @MaximAlien  thanks for your help , actually yes we are using `react-native-mapbox-gl` in our project in many areas, is there any another way to achieve the successfully upgrade to latest version without removing ` react-native-mapbox-gl`

**Comment by MaximAlien** [comment]
@zreaqqasem, can you briefly describe your use case? What's the reason you're using `react-native-mapbox-gl` and link `MapboxNavigation` at the same time? What features of Mapbox Navigation you currently use? 

Since `react-native-mapbox-gl` library is for cross-platform mobile development it doesn't align with Mapbox Navigation SDK for iOS anyway, as it can be purely used on iOS, so you'll not be able to ship it for multiple platforms (or I'm missing something).

So at this point I do not see any ways of fixing this other than dropping `react-native-mapbox-gl` usage completely in your application, or upgrading it to the latest version of Mapbox Maps SDK `v10.0.x`.

**Comment by zreaqqasem** [comment]
@MaximAlien thank you again for your your reply, maybe it’s not clear from my issue what type of application we use, actually yes we are developing a cross platform mobile app using react-native and we are using `react-native-mapbox-gl` In the our reqct-native-component, at the same time we have a native module written in swift and we are using there `MapBoxNavigation` and everything works as expected but once we are decide to upgrade our project we stucked in this issue. What’s your advice? Keep on `0.4.0 ` version or there’s an alternative solution?

**Comment by MaximAlien** [comment]
@zreaqqasem I assume you're using one Xcode project for everything in this case? What about extracting target, which is using `MapboxNavigation` into separate project, so that it can work independently? Or those targets depend on each other heavily?

This is just more of a workaround, best solution in this case would be to upgrade `react-native-mapbox-gl` to Mapbox Maps `v10.x`, I've never worked with this library though, so cannot estimate possible effort for this.

**Comment by MaximAlien** [comment]
Linking https://github.com/react-native-mapbox-gl/maps/discussions/1202 and https://github.com/react-native-mapbox-gl/maps/pull/1222, it might be useful for you. PR is a bit outdated though. Maps SDK development was in active development at the time when it was created and lots of changes were introduced since then.

**Comment by zreaqqasem** [comment]
> @zreaqqasem I assume you're using one Xcode project for everything in this case? What about extracting target, which is using `MapboxNavigation` into separate project, so that it can work independently? Or those targets depend on each other heavily?
> 
> This is just more of a workaround, best solution in this case would be to upgrade `react-native-mapbox-gl` to Mapbox Maps `v10.x`, I've never worked with this library though, so cannot estimate possible effort for this.

yes we use one Xcode project, it's a one project, the idea is as the following -> we are using    
` "@mapbox/mapbox-sdk": "^0.13.2"` and  `"@react-native-mapbox-gl/maps": "^8.5.0"` in a react native and use this packages, in the other hand in the navigation side we are go with write that side using a native components so we used there in our swift native module  ` pod 'MapboxNavigation', '~> 0.40.0'` when we come to upgrade our project and packages, we are succelfuly did that in the android side, but here in iOS side we faced this issue.
